### PR TITLE
FE-1272 Add Messaging Usage Chart to Usage Page

### PR DESCRIPTION
**Pull request by logansparlin**
### What Changed

- Adds `AreaChart` component
- Adds MessagingUsageChart to Usage Page

### How To Test

- Run the app
- `npm run test-integration-hibana` and run the accounts > usage page integration tests
  > Visiting this page in the app will throw an error until [AC-1809](https://sparkpost.atlassian.net/browse/AC-1809) is resolved
### To Do

- [ ] Any outstanding tasks go here
  > Manage reviewers' expectations on missing features, tests, bugs, late design changes, ...)

### CODE REVIEW

File: src/pages/usage/UsagePage.js [hunk 1]
```diff
@@ -6,27 +6,39 @@ import {
...
+    accountStatus === 'loading' ||
+    usageStatus === 'loading' ||
+    subscriptionStatus === 'loading' ||
+    usageHistoryStatus === 'loading';
   const isError =
-    accountStatus === 'error' || usageStatus === 'error' || subscriptionStatus === 'error';
+    accountStatus === 'error' ||
+    usageStatus === 'error' ||
+    subscriptionStatus === 'error' ||
+    usageHistoryStatus === 'error';
```
Comment (bkemper) [inline src/pages/usage/UsagePage.js:36]: :) Looks like a little overlap with #1935

File: src/pages/usage/components/MessagingUsageChart.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        <Box key="report_chart_dailylimit" mb="100">
+          <Box justifyContent="space-between" alignItems="center" display="flex">
+            <Box display="inline-flex" alignItems="center">
+              <Text as="span" fontSize="100" color="white">
+                Daily Limit
+              </Text>
+            </Box>
+            <Box ml="800">
+              <Text fontSize="100" textAlign="right" color="white">
+                {usage.day.limit.toLocaleString()}
```
Comment (bkemper) [inline src/pages/usage/components/MessagingUsageChart.js:51]: This `usage.day.limit.toLocaleString()` line is failing for me with appteam account.  Judging by how `MessagingUsageSection` guards all use of `usage.day.limit` with `hasDailyLimit`, this must be a valid use case — an account without limits.  It is not documented in the API docs, so I'll check with AC team.

Screen Shot 2020-12-10 at 8 54 48 AM

File: src/pages/usage/components/tests/MessagingUsageChart.test.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import MessagingUsageChart from '../MessagingUsageChart';
+
+describe('MessagingUsageChart', () => {
```
Comment (bkemper) [inline src/pages/usage/components/tests/MessagingUsageChart.test.js:5]: Can we move this test to Cypress?

File: src/pages/usage/components/MessagingUsageChart.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+      </defs>
+    );
+  }
+
+  return (
+    <div>
+      <AreaChart
+        height={200}
+        areas={[{ dataKey: 'usage', stroke: 'url(#strokeColor)', fill: 'url(#fillColor)' }]}
+        data={data}
```
Comment (bkemper) [inline src/pages/usage/components/MessagingUsageChart.js:98]: I confirmed that even with RV accounts, there should be at least one datapoint, but should we populate the rest of the chart?

Screen Shot 2020-12-10 at 9 18 42 AM

File: src/components/charts/AreaChart.js [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+            height={30}
+            hide={!showXAxis}
+            interval="preserveStartEnd"
+            tickFormatter={xTickFormatter}
+            tickLine={false}
+            ticks={getXTicks()}
+          />
+          <YAxis
+            axisLine={false}
+            domain={['dataMin', 'dataMax']}
```
Comment (bkemper) [inline src/components/charts/AreaChart.js:124]: I think we always want the reference line to be visible.  Using `dataMax` will only show the reference line if it exceeds the limit.  Instead, what you do you think about doing something like this?

```
domain={['dataMin', dataMax => Math.max(dataMax, ...yAxisRefLines.map(({ y }) => y))]}
```

**Review by bkemper — COMMENTED** [review]
Looks good!  I left a few comments.  Some of the feedback overlaps with Jon's PR.

**Comment by shrirupa** [comment]
I think the failing integration test is because we might have to add the stub request for usage history on useSparkPostQuery.spec.js

File: src/pages/usage/components/MessagingUsageChart.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+import { Box, Text } from 'src/components/matchbox';
+import { tokens } from '@sparkpost/design-tokens-hibana';
+import { formatNumber } from 'src/helpers/units';
+import { fillByDate } from 'src/helpers/date';
+import { getTimeTickFormatter, getTooltipLabelFormatter } from 'src/helpers/chart.js';
+
+function MessagingUsageChart(props) {
+  const { data = [], planVolume, overage, dailyLimit, start, end } = props;
+
+  const filledData = React.useMemo(() => {
```
Comment (shrirupa) [inline src/pages/usage/components/MessagingUsageChart.js:12]: I think for the usage we want to display the cumulative usage for each day.

File: cypress/tests/integration/accounts/usage/usagePage.spec.js [hunk 7]
```diff
@@ -130,5 +130,24 @@ describe('The usage page', () => {
...
+      cy.stubRequest({
+        url: '/api/v1/usage/history',
+        fixture: 'usage/history/200.get.json',
+        requestAlias: 'usageHistoryReq',
+      });
+
+      cy.visit(PAGE_URL);
+      cy.wait(['@usageReq', '@usageHistoryReq', '@subscriptionReq', '@accountReq']);
+      cy.findByText('Dec 29th').should('be.visible');
+      cy.findByText('80K').should('be.visible');
```
Comment (shrirupa) [inline cypress/tests/integration/accounts/usage/usagePage.spec.js:150]: this line seems to be failing locally too.

**Review by shrirupa — APPROVED** [review]
Looks good to me! I think for the cypress test to check if the chart has rendered or not we can add a data-id to messaging chart and check that in the test.

**Pull request merged**